### PR TITLE
241872: Azure monitor profile fix and test

### LINKED ISSUE #4046: [Feature Request]: Monitoring workspace for AKS Cluster
### Description

Monitoring workspace module which is a pre-requisite for Managed Promitheus deployment for AKS Cluster is not yet available on Azure ResourceModules repository ( container/managed cluster ). 

dependencies.bicep file:
```bicep
resource monitoringWorkspace 'Microsoft.Monitor/accounts@2023-04-03' = {
  name: monitoringWorkspaceName
  location: location
}
```
main.bicep file: 
```bicep
monitoringWorkspaceId: nestedDependencies.outputs.workspaceResourceId
```
This is missing, hence this feature request.

### PR DESCRIPTION
**Pull request by osagiestar**
# Description

Add MonitorProfile property required for enabling Prometheus on AKS

<details>
<summary>Issues that this PR fixes</summary>
Fixes #4046
</details>

-->

### Pipeline references

| Pipeline |
| - |
| |

## Type of Change

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/container-service/managed-cluster/main.bicep [hunk 1]
```diff
@@ -389,9 +404,18 @@ param diagnosticMetricsToEnable array = [
   'AllMetrics'
 ]
 
-@description('Optional. The name of the diagnostic setting, if deployed. If left empty, it defaults to "<resourceName>-diagnosticSettings".')
+@description('Required. The name of the diagnostic setting, if deployed. If left empty, it defaults to "<resourceName>-diagnosticSettings".')
```
Comment (AlexanderSehr) [inline modules/container-service/managed-cluster/main.bicep:407]:
```suggestion
@description('Optional. The name of the diagnostic setting, if deployed. If left empty, it defaults to "<resourceName>-diagnosticSettings".')
```

File: modules/container-service/managed-cluster/.test/workspace/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.containerservice.managedclusters-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'csmaz'
```
Comment (AlexanderSehr) [inline modules/container-service/managed-cluster/.test/workspace/main.test.bicep:15]:
```suggestion
param serviceShort string = 'csw'
```

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Nice PR @osagiestar ,
added a few suggestions and would have 1 question: Have you been able to successfully validate the changes in your fork?
Ideally, for these kind of changes we ask contributors to provide a pipeline badge to a successful run so we can be sure it works. However, that requires you to have a subscription, and we can not always expect that :) If that's not an option we'll find another way to test.

**Comment by osagiestar** [comment]
> Nice PR @osagiestar , added a few suggestions and would have 1 question: Have you been able to successfully validate the changes in your fork? Ideally, for these kind of changes we ask contributors to provide a pipeline badge to a successful run so we can be sure it works. However, that requires you to have a subscription, and we can not always expect that :) If that's not an option we'll find another way to test.

Please go ahead and test @AlexanderSehr

**Comment by AlexanderSehr** [comment]
Hey @osagiestar will do and merge your contribution soon. However, to do so I'd need to ask you to please resolve the conflicts first by merging upstream CARML back into your fork. It appears something in the AKS module changed in the meantime that conflicts now. If that turns out to be big a mess (Git can get quite complicated) I can also port the changes over manually, but this would mean that the history is lost :/

**Comment by PixelRobots** [comment]
Any update on this?

**Comment by AlexanderSehr** [comment]
cc: @JPEasier please include this one in the AVM module too (if not already considered)

**Comment by AlexanderSehr** [comment]
@osagiestar any chance on resolving the conflicts? Again, otherwise I can port the changes from our end, but will only do so with your go ahead.

**Comment by osagiestar** [comment]
> @osagiestar any chance on resolving the conflicts? Again, otherwise I can port the changes from our end, but will only do so with your go ahead.

@AlexanderSehr - If it has not been resolved yet and you can resolve it from your end that will be great. Regards.

**Comment by AlexanderSehr** [comment]
Alright, this was merged as part of PR https://github.com/Azure/bicep-registry-modules/pull/665 and will hence be closed. Thanks  @osagiestar for your contribution 💪

**Pull request closed without merging**